py/farm_ng/core/event_service.py

```diff
@@ -431,13 +433,29 @@ async def _send_raw(
             sequence=count,
         )
 
+        num_clients: int = await self._publish_event_payload(
+            event=event,
+            payload=payload,
+            latch=latch,
+        )
+
+        # update the metrics with the count of the clients the message was sent to
+        uri_path_send_clients: str = f"{uri.path}/send_clients"
+        if uri_path_send_clients not in self._metrics.data:
+            self._metrics.data[uri_path_send_clients] = 0
+
+        self._metrics.data[uri_path_send_clients] += num_clients
+
+        # update the metrics with the timestamp of the last message
+        # to compute the rate of the messages.
+        if uri.path not in self._metrics.data_tmp:
+            self._metrics.data_tmp[uri.path] = []
+
+        self._metrics.data_tmp[uri.path].append(timestamps[-1].stamp)
```

Comment:
Suggest to use [`get_stamp_by_semantics_and_clock_type`](https://github.com/farm-ng/farm-ng-core/blob/main/py/farm_ng/core/stamp.py#L68) rather than indexing by order in a lsit

Reply:
done